modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointPagesWriter.java

```diff
@@ -240,35 +247,25 @@ public void writePage(FullPageId fullPageId, ByteBuffer buf, int tag) throws Ign
         };
     }
 
-    /**
-     * Updates the partitions meta page.
-     *
-     * @throws IgniteInternalCheckedException If failed.
-     */
-    private void updatePartitions() throws IgniteInternalCheckedException {
-        Result<PersistentPageMemory, GroupPartitionId> queueResult = new Result<>();
+    private void writePartitionMeta(GroupPartitionId partitionId, ByteBuffer buffer) throws IgniteInternalCheckedException {
+        PartitionMetaSnapshot partitionMetaSnapshot = partitionMetaManager.getMeta(partitionId).metaSnapshot(checkpointProgress.id());
 
-        while (!shutdownNow.getAsBoolean() && updatePartitionIds.next(queueResult)) {
-            GroupPartitionId partitionId = queueResult.getValue();
+        partitionMetaManager.writeMetaToBuffer(partitionId, partitionMetaSnapshot, buffer.rewind());
 
-            ByteBuffer buffer = threadBuf.get();
+        FullPageId fullPageId = new FullPageId(partitionMetaPageId(partitionId.getPartitionId()), partitionId.getGroupId());
 
-            PersistentPageMemory pageMemory = queueResult.getKey();
-
-            PartitionMetaSnapshot partitionMetaSnapshot = partitionMetaManager.getMeta(partitionId).metaSnapshot(checkpointProgress.id());
-
-            partitionMetaManager.writeMetaToBuffer(partitionId, pageMemory, partitionMetaSnapshot, buffer.rewind());
+        PageStore store = pageWriter.write(fullPageId, buffer.rewind());
 
-            FullPageId fullPageId = new FullPageId(
-                    pageMemory.partitionMetaPageId(partitionId.getGroupId(), partitionId.getPartitionId()),
-                    partitionId.getGroupId()
-            );
+        checkpointProgress.writtenPagesCounter().incrementAndGet();
 
-            PageStore store = pageWriter.write(fullPageId, buffer.rewind());
+        updStores.computeIfAbsent(store, k -> new LongAdder()).increment();
+    }
 
-            checkpointProgress.writtenPagesCounter().incrementAndGet();
+    private static boolean isPartitionChange(@Nullable GroupPartitionId partitionId, FullPageId pageId) {
```

Comment:
Sorry for making comments about naming, but how should I interpret this name?

Reply:
Has the partition changed?

Reply:
Why didn't you call it "hasPartitionChanged"?
"is partition change" looks weird

Reply:
Fix it